server/neptune/sync/exec.go

```diff
@@ -0,0 +1,70 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	key "github.com/runatlantis/atlantis/server/neptune/context"
+	"github.com/runatlantis/atlantis/server/neptune/logger"
+	"os"
+	"os/exec"
+	"syscall"
+	"time"
+)
+
+// RunNewProcessGroupCommand is useful for running separate commands that shouldn't receive termination
+// signals at the same time as the parent process
+func RunNewProcessGroupCommand(ctx context.Context, cmd *exec.Cmd, cmdName string) error {
```

Comment:
Are we confidant that all the callers of this fun pass in a context that allows us to gracefully drain for 60min or whatever time were configuring?

Reply:
If you are asking if the current function callers use a correct context, I believe so: 
The two users here (HooksRunner and RepoCloner) are using the context provided the AsyncScheduler that wraps the relevant GitHub event workflows into asynchronous go routines [here](https://github.com/lyft/atlantis/blob/release-v0.17.3-lyft.1/server/neptune/sync/scheduler.go#L40-L64). While doing so, the original context is replaced with a shared parent context that has its [own](https://github.com/lyft/atlantis/blob/release-v0.17.3-lyft.1/server/neptune/gateway/server.go#L426) 30 second shutdown timeout. That shutdown call won’t happen until we [receive](https://github.com/lyft/atlantis/blob/release-v0.17.3-lyft.1/server/neptune/gateway/server.go#L406) the original SIGTERM to shutdown. Note that this is gateway, which doesn't perform Terraform processes so doesn't have the 60 minute timeout.

If you are asking if the function guarantees future use cases correctly use a context that listens for a graceful timeout:
No, similar to my older comment on whether or not the function’s intentions are clear or not, technically another code change could pass a context that doesn’t listen for it. I can add comments to try to prevent this.